source/common/filter/config_discovery_impl.h

```diff
@@ -173,40 +172,42 @@ struct NamedHttpFilterFactoryCb {
 };
 
 // Implementation of a HTTP dynamic filter config provider.
+template <class FactoryCtx, class NeutralHttpFilterConfigFactory>
```

Comment:
Should Neutral be `Named`? Or this is a new concept introduced?

Reply:
It's upstream/downstream neutral.  Can rename to UpstreamOrDownstreamHttpFilterConfigFactory if that's more clear?

Reply:
Neutral is fine with an explanation. Maybe a comment on top so it's possible to tell what this is without paying the cost of huge type name?